insights/client/apps/compliance/__init__.py

```diff
@@ -32,13 +34,18 @@ def oscap_scan(self):
         if not profiles:
             logger.error("System is not associated with any profiles. Assign profiles using the Compliance web UI.\n")
             exit(constants.sig_kill_bad)
+
+        archive_dir = self.archive.create_archive_dir()
         for profile in profiles:
+            tailoring_file = self.download_tailoring_file(profile)
             self.run_scan(
                 profile['attributes']['ref_id'],
                 self.find_scap_policy(profile['attributes']['ref_id']),
-                '/var/tmp/oscap_results-{0}.xml'.format(profile['attributes']['ref_id']),
-                tailoring_file_path=self.download_tailoring_file(profile)
+                self._results_file(archive_dir, profile),
+                tailoring_file_path=tailoring_file
             )
+            if tailoring_file:
+                os.unlink(tailoring_file)
```

Comment:
`unlink` and `remove` seem identical? The [os.unlink docs](https://docs.python.org/3/library/os.html#os.unlink) state:

>This function is semantically identical to remove(); the unlink name is its traditional Unix name.

Nit: IMO `remove` makes it more readable for those who aren't as familiar with Python (or UNIX).